api/client/service/opts.go

```diff
@@ -253,6 +307,21 @@ func (m *MountOpt) Value() []swarm.Mount {
 	return m.values
 }
 
+func getMountType(fields []string) (swarm.MountType, []string) {
```

Comment:
@dnephin The original grammar used a positional sentinel value to figure out how to parse the rest of the syntax. This sentinel changed with the addition of windows support. The complex arose when having to backtrack on that sentinel.

This is the same situation, which may lead to the future of additional backtracking.

Making each expression be considered independently keeps the lexicographic parsing, rewrites and ultimately structure separate and will yield a better result over time.